File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        N, area = len(height), 0
        beg, end = 0, N - 1
        
        while beg < end:
            area = max(area, (min(height[beg], height[end])*(end-beg)))
            if height[beg] < height[end]:
                beg += 1
            else:
                end -= 1
        return area

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int beg = 0, end = n - 1, ans = 0;
        while (beg < end) {
            ans = max(ans, (min(height[end], height[beg]) * (end - beg)));
            if (height[beg] < height[end])
                beg++;
            else
                end--;
        }
        return ans;
    }
};